File: replace_section.js
const fs = require('fs')
const path = 'src/app/menu/quadrants/[id]/components/QuadrantModal.tsx'
const text = fs.readFileSync(path, 'utf8')
const startMarker = '          {/* Camps específics */}'
const endMarker = '          {department.toLowerCase() ===  logistica && ('
const start = text.indexOf(startMarker)
const end = text.indexOf(endMarker, start)
if (start === -1 || end === -1) {
  throw new Error('markers not found')
}
const newBlock =           {/* Department-specific sections */}\n          {isServeis && (\n            <AutoGenerateServeis\n              serveisTotals={serveisTotals}\n              serveisGroups={serveisGroups}\n              addServeisGroup={addServeisGroup}\n              updateServeisGroup={updateServeisGroup}\n              removeServeisGroup={removeServeisGroup}\n              availableConductors={conductors}\n              eventStartDate={startDate}\n            />\n          )}\n
fs.writeFileSync(path, text.slice(0, start) + newBlock + text.slice(end), 'utf8')

File: replace_section.py
from pathlib import Path
path = Path('src/app/menu/quadrants/[id]/components/QuadrantModal.tsx')
text = path.read_text()
start = text.index('          {/* Camps específics */}')
end = text.index('          {department.toLowerCase() === " logistica\ && (', start)
new = ''' {/* Department-specific sections */}
 {isServeis && (
 <AutoGenerateServeis
 serveisTotals={serveisTotals}
 serveisGroups={serveisGroups}
 addServeisGroup={addServeisGroup}
 updateServeisGroup={updateServeisGroup}
 removeServeisGroup={removeServeisGroup}
 availableConductors={conductors}
 eventStartDate={startDate}
 />
 )}
'''
text = text[:start] + new + text[end:]
path.write_text(text)
